.github/workflows/macos.yml

```diff
@@ -71,9 +84,36 @@ jobs:
 
       - run: make
 
+      # Launchable requires Python and Java
+      # https://www.launchableinc.com/docs/resources/cli-reference/
+      - name: Set up Python
+        uses: actions/setup-python@871daa956ca9ea99f3c3e30acb424b7960676734 # v5.0.0
```

Comment:
What is the reason to set the commit hash `871daa956ca9ea99f3c3e30acb424b7960676734` here? Seeing the https://github.com/actions/setup-python/commits/v5.0.0/, the `871daa956ca9ea99f3c3e30acb424b7960676734` is not the same with the `v5.0.0` or `v5` tag but it is the commit on the current latest main branch. Why we don't do the `uses: actions/setup-python@v5.0.0` or `uses: actions/setup-python@v5`?

Reply:
GitHub recommends using commit SHAs to specify third-party versions to prevent unintentional changes that could occur if tags are updated later. This approach ensures stability and avoids potential issues with dependency updates. For example, specifying a tag like `v5.0.0` may lead to unintended changes if the tag is updated in the future.

In addition to that, the reason of specifying the latest commit hash here is because I checked the latest commit hash and there is no dangerous change in this commit hash.

Reply:
Yes, I can.
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
Thanks for the reference source! Okay. I learned. Below is the citation as my note, and for someone who is interested in this topic.

> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload.